plugins/kotlin-as-java/src/main/kotlin/KotlinAsJavaPageBuilder.kt

```diff
@@ -0,0 +1,132 @@
+package org.jetbrains.dokka.kotlinAsJava
+
+import org.jetbrains.dokka.kotlinAsJava.conversions.asJava
+import org.jetbrains.dokka.kotlinAsJava.conversions.asStatic
+import org.jetbrains.dokka.kotlinAsJava.conversions.withClass
+import org.jetbrains.dokka.links.withClass
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.Function
+import org.jetbrains.dokka.model.doc.TagWrapper
+import org.jetbrains.dokka.pages.*
+import org.jetbrains.dokka.transformers.descriptors.KotlinClassKindTypes
+import org.jetbrains.kotlin.descriptors.Visibilities
+
+class KotlinAsJavaPageBuilder(val rootContentGroup: RootContentBuilder) {
+    fun pageForModule(m: Module): ModulePageNode =
+        ModulePageNode(m.name.ifEmpty { "root" }, contentForModule(m), m, m.packages.map { pageForPackage(it) })
+
+    data class FunsAndProps(val key: String, val funs: List<Function>, val props: List<Property>)
+
+    private fun pageForPackage(p: Package): PackagePageNode {
+
+        val funs = p.functions.filter { it.sourceLocation != null }.groupBy { function ->
+            function.sourceLocation!!.let { it.split("/").last().split(".").first() + "Kt" }
```

Comment:
`sourceLocation` can be blank for synthetic Documentables, this will fails with an exception

Reply:
Now it's extracted directly from descriptor.